File: integration_tests/java/discriminator/V1.java
package com.jsontypedef.jtdcodegendemo;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonTypeInfo;
import com.fasterxml.jackson.annotation.JsonValue;

import java.util.List;
import java.util.Map;


public class V1 extends Discriminator {

  
  @JsonProperty("user")
  private V1User user;


  
  public V1() {
  }
  


  public V1User getUser() {
    return user;
  }

  public void setUser(V1User user) {
    this.user = user;
  }

}

File: integration_tests/csharp/discriminator/V1.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;

namespace Jtd.JtdCodegenDemo
{

    
    public class V1 : Discriminator
    {

        [JsonProperty("version")]
        public const string Version = "v1";



        [JsonProperty("user")]
        public V1User User { get; set; }





    }
}
